.github/workflows/build.yml

```diff
@@ -31,13 +33,14 @@ jobs:
         uses: docker/build-push-action@v5
         with:
           context: .
-          push: true
+          push: ${{ github.event_name == 'push' }} # Only push to PRD if the event is an push event
```

Comment:
Will give this a full review when I'm off work, but curious what you're trying to achieve here? 🙂 

Every single branch (and every commit on every branch) is intentionally pushed up as we have branch deployments. This allows us to see what the website will look like on every PR, before we merge it (and without running locally).

Reply:
![image](https://github.com/SwiftLeeds/swiftleeds-web/assets/1290461/53b4767b-fca2-4eba-b72e-92c2bedba482)

People not in the SwiftLeeds org cannot use the runner with the secret, so for every outside collaborator (e.g. me), it will fail with the above error.

i tried to avoid that error.

Reply:
Got that part, but is that not fixed by only running the workflow where the repository owner is 'SwiftLeeds'?

That seems acceptable, but changing the actual scripts to only push to GCP on merge into `main` fundamentally changes our workflow which would not be accepted here.

I'll also add that you can simply turn off GitHub Actions in your fork settings 🙂

Reply:
True, but by turning off GitHub actions in the fork, then when you do the PR it does not work, in general the Repo owner = SwiftLeeds (or any other) is smart for runs which relies on secrets.

I think it is better then to change `push: ${{ github.event_name == 'push' }}` to something what checks if you can access the secrets, so something like `push: ${{ secrets.GAR_JSON_KEY != '' }}` (not tested btw)

Reply:
Actions will still be run when you open a PR here, even if you turn them off on your fork - and again, I do not disagree with the idea of checking for repo owner = SwiftLeeds (I'd be happy to merge that change).

However, I do not see the value being obtained by disabling both the job (by virtue of the repo owner check) and the steps (by checking the event name). If the job isn't running, it doesn't matter whether the secrets exist. The repo owner check should be enough on it's own.

Reply:
I do agree that it _should_ be enough, but if i see the tet result it still fails, https://github.com/SwiftLeeds/swiftleeds-web/actions/runs/9302012857/job/25601329359

I'll revert the `if: ...` on the steps, then at least people with forks don't get a test failed message if they fork :)

Reply:
I hear you now (the link helps with context), but feel we need to solve that in a different way.

What we can't do is disable key functionality from our workflows (i.e. the ability to deploy branches). Need to better understand GHA permissions here.

It might be that for branch deployments we need to change the workflow so it only runs when the parent branch is owned by SwiftLeeds (`if: ${{ github.event.pull_request.head.repo.full_name == github.event.pull_request.base.repo.full_name }}`). This would allow us to (in the future) still run tests.

Reply:
i found the solution to the issue ( https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/ )

If we change `pull_request` to `pull_request_target` it will run (with the secrets) after (someone in the org) accepted to run the action.

Reply:
This sounds like an excellent solution. Let's try that!

Reply:
Done :)